Source/Engine/Editor/VQAssetBrowser.cpp

```diff
@@ -0,0 +1,291 @@
+//	VQE
+//	Copyright(C) 2020  - Volkan Ilbeyli, Anilcan Gulkaya
+//
+//	This program is free software : you can redistribute it and / or modify
+//	it under the terms of the GNU General Public License as published by
+//	the Free Software Foundation, either version 3 of the License, or
+//	(at your option) any later version.
+//
+//	This program is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+//	GNU General Public License for more details.
+//
+//	You should have received a copy of the GNU General Public License
+//	along with this program.If not, see <http://www.gnu.org/licenses/>.
+//
+//	Contact: volkanilbeyli@gmail.com, anilcangulkaya7@gmail.com
+
+#include "VQAssetBrowser.h"
+#include <vector>
+#include <string>
+#include <fstream>
+#include <filesystem>
+#include "../Core/Types.h"
+#include "../../Libs/VQUtils/Source/Log.h"
+
+CBV_SRV_UAV* AssetBrowser::ExtensionToIcon(std::string& extension) const
```

Comment:
No need for the hasher class -- can use an unordered_map<string, TexSRVPair> instead.

Reply:
I think no need for the unordered_map because it's memory heavy. I now unordered_map::find has O(1) time complexity, we only going to use maximum 10 extension types (now we use 3 types) so comparing with switch case is fast O(1). You know better than me :) if you have another reason to not use hash I can gladly change this.

Reply:
> I think no need for the unordered_map because it's memory heavy

How much memory do you intend to save by not using the unordered_map?
My estimate is 10 strings * 4-6 characters each ~ 60 Bytes.
I wouldn't worry too much for saving 60 bytes in this section of the engine.

> I know unordered_map::find has O(1) time complexity but we only going to use maximum 10 extension types (now we use 3 types) so comparing with switch case is fast O(1). 

time complexity is relevant for large inputs -- as you mentioned we don't have a large input scenario so the time complexity doesn't matter when deciding which container to use for this particular 

case. So there is another reason, and that is: keep it simple.

Each newly added functionality increases the probability of a bug, hence each addition must be justified.
As we're not worried about saving ~60bytes, and we're not dealing with long string, and we're not dealing with a large number of strings, a simple container like unordered_map will do the trick just fine.